server/etcdserver/api/etcdhttp/base.go

```diff
@@ -40,24 +43,46 @@ func HandleBasic(lg *zap.Logger, mux *http.ServeMux, server etcdserver.ServerPee
 	mux.HandleFunc(versionPath, versionHandler(server.Cluster(), serveVersion))
 }
 
-func versionHandler(c api.Cluster, fn func(http.ResponseWriter, *http.Request, string)) http.HandlerFunc {
+func HandleBasicForV3(lg *zap.Logger, mux *http.ServeMux, server serverversion.Server) {
```

Comment:
I don't it makes sense to ditinquish between v2 and v3 API as V2 is officially removed in next release. We should be cleaning up V2 code and removing mention of separation like `V3` suffix in names.

Reply:
Yes, I agree. But can we refactor the name & code after v2 is officially removed?

Reply:
V2 should be already removed.

Reply:
It seems that v2 isn't completed removed. If you open [server/etcdserver/api](https://github.com/etcd-io/etcd/tree/main/server/etcdserver/api), you will see a couple of folders has prefix v2. This function [HandleBasic](https://github.com/etcd-io/etcd/blob/91f9ce9f5b52ab1c778c129dc102525859baee93/server/etcdserver/api/etcdhttp/base.go#L38) is exactly used by one of the v2 package [v2http](https://github.com/etcd-io/etcd/blob/91f9ce9f5b52ab1c778c129dc102525859baee93/server/etcdserver/api/v2http/client.go#L33). 

I also see that you have a similar name [HandleMetricsHealthForV3](https://github.com/etcd-io/etcd/blob/91f9ce9f5b52ab1c778c129dc102525859baee93/server/embed/etcd.go#L705), so I followed the same name pattern so as to be consistent. 

Anyway, let's refactor the name in a separate PR. WDYT?

Reply:
+1 to @serathius and I am good with the plan of a clean sweep on v2. Also, as we refactor, it's a good practice to add doc for the public functions and so to keep it in mind when we add new ones. Thanks @ahrtr !!

Reply:
A note: There is another PR [pull/13655](https://github.com/etcd-io/etcd/pull/13655) for the cleanup of v2.

Reply:
Resolved.